src/server/req_manager.c

```diff
@@ -1633,9 +1633,12 @@ mgr_server_unset(struct batch_request *preq)
 				tm_list->al_link.ll_next->ll_struct = NULL;
 				rc = mgr_unset_attr(dflt_scheduler->sch_attr, sched_attr_def, SCHED_ATR_LAST, tm_list,
 					-1, &bad_attr, (void *)dflt_scheduler, PARENT_TYPE_SCHED, INDIRECT_RES_CHECK);
-				if (rc != 0)
+				if (rc != 0) {
+					free_svrattrl(tm_list);
 					reply_badattr(rc, bad_attr, plist, preq);
+				}
 				(void)sched_save_db(dflt_scheduler, SVR_SAVE_FULL);
+				free_svrattrl(tm_list);
```

Comment:
In the case of rc != 0, you're freeing tm_list twice.

Reply:
Done